File: javascript/762.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */

function isPrime(n) {
    if(n <= 1) return false;

    for (let i = 2; i <= Math.sqrt(n); i++){
        if (n % i == 0) return false;
    }

    return true;
}
var countPrimeSetBits = function(left, right) {
    let total = 0;

    for (let i = left; i <= right; i++) {
        let bits = ((i >>> 0).toString(2).match(/1/g) || []).length;
        if (isPrime(bits)) ++total;
    }

    return total;
};

File: kotlin/762.kt
// Código 1

class Solution {
    fun countPrimeSetBits(left: Int, right: Int): Int {
        var total = 0

        for (i in left until right + 1) {
            val bits = i.countOneBits()
            if (isPrime(bits)) ++total
        }

        return total
    }

    fun isPrime(n: Int): Boolean {
        if (n <= 1) {
            return false
        }
    
        for (i in 2 until Math.sqrt(n.toDouble()).toInt() + 1) {
            if (n % i == 0) {
                return false
            }
        }
    
        return true
    } 
}


// Código 2

class Solution {
    fun countPrimeSetBits(left: Int, right: Int): Int {
        var total = 0

        for (i in left until right + 1) {
            val bits = i.countOneBits()
            if (isPrime(bits)) ++total
        }

        return total
    }

    fun isPrime(n: Int): Boolean {
        if (n <= 1) {
            return false
        }
    
        for (i in 2 until Math.sqrt(n.toDouble()).toInt() + 1) {
            if (n % i == 0) {
                return false
            }
        }
    
        return true
    } 
}

// Código 3

class Solution {
    fun countPrimeSetBits(left: Int, right: Int): Int {
        var total = 0

        for (i in left until right + 1) {
            val bits = i.countOneBits()
            if (isPrime(bits)) ++total
        }

        return total
    }

    fun isPrime(n: Int): Boolean {
        if (n <= 1) {
            return false
        }
    
        for (i in 2 until Math.sqrt(n.toDouble()).toInt() + 1) {
            if (n % i == 0) {
                return false
            }
        }
    
        return true
    } 
}